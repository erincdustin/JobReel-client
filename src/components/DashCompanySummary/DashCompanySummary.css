.DashCompanySummary > h4 {
  text-align: center;
}

.options > ul > li {
  list-style: none;
  width: 100%;
}

.ul {
  padding-left: 0;
  width: 95%;
  margin: auto;
  height: 100%;
}

.viewAll {
  text-align: center;
}