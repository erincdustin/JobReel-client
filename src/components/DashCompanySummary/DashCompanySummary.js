import React from 'react';
import { Link } from 'react-router-dom';
import './DashCompanySummary.css';
import jobReelContext from '../../context/JobReelContext';
import jobReelApiService from '../../services/jobreel-api-service';

class DashCompanySummary extends React.Component {

  static contextType = jobReelContext

  componentDidMount() {
    if (this.context.companies.length === 0) {
      jobReelApiService.getSavedCompanies()
        .then(res => {
          this.context.setCompanies(res);
        })
    }
  }

  renderCompanySummaries = () => {
    let companies = Array.from(this.context.companies);
    companies = companies.sort((a, b) => {
      return new Date(b.date_added) - new Date(a.date_added);
    });
    companies = companies.slice(0, 3).map(company => {
      return (
        <li key={company.company_id}>{company.company_name} - <a href={company.website}>{company.website}</a>- Contact: {company.contact}</li>
      )
    })
    return (
      <ul>
        {companies}
      </ul>
    )
  }
  
  render() {
    return (
      <>
        <div className="DashCompanySummary">
          <h4>{this.props.section}</h4>
        </div>
        <div className='options'>
            {this.renderCompanySummaries()}
        </div>
        <Link to="/companies">View all</Link>
      </>
    )
  }
}

export default DashCompanySummary;