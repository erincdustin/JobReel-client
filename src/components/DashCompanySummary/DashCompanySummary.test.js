import React from 'react';
import ReactDOM from 'react-dom'
import {shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import { BrowserRouter } from 'react-router-dom'
import DashCompanySummary from './DashCompanySummary'

class ContextProvider extends React.Component {
  getChildContext = () => ({
    myContext: {
      test: 2,
      companies: [1, 2, 3]
    }
  })

  render(){
    return this.props.children
  }
}
describe('DashCompanySummary smoke test', () => {
  it('renders without crashing', () => {
      const div = document.createElement('div');
      ReactDOM.render(<BrowserRouter><DashCompanySummary /></BrowserRouter>, div);
      ReactDOM.unmountComponentAtNode(div);
  })
})

describe('DashCompanySummary snapshot test', () => {
  it('renders the UI as expected', () => {
    const context = { companies: ['foo']}
    const wrapper = shallow(<ContextProvider><DashCompanySummary /></ContextProvider>, { context })
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})