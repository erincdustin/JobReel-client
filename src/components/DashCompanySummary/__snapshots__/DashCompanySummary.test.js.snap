// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DashCompanySummary snapshot test renders the UI as expected 1`] = `<DashCompanySummary />`;
